main.tf

```
@@ -46,3 +46,41 @@ module "s3_user" {
   s3_actions   = ["${var.allowed_bucket_actions}"]
   s3_resources = ["${join("", aws_s3_bucket.default.*.arn)}/*", "${join("", aws_s3_bucket.default.*.arn)}"]
 }
+
+resource "aws_s3_bucket_policy" "default" {
+  count  = "${var.allow_encrypted_uploads_only == "true" ? 1 : 0}"
```

```suggestion
  count  = "${var.enabled == "true" && var.allow_encrypted_uploads_only == "true" ? 1 : 0}"
```

If `var.enabled == "false"`, all resources should be disabled, including `aws_s3_bucket_policy`

---

done